adsenrich/references.py

```diff
@@ -50,24 +70,50 @@ def _create_output_file_name(self):
             if not self.data:
                 raise MissingDataException("You have not provided a record to process.")
             if not self.bibcode:
-                self.bibcode = BibcodeGenerator().make_bibcode(self.data)
+                self.bibcode = BibcodeGenerator(
+                    token=self.api_token, url=self.api_url
+                ).make_bibcode(self.data)
                 if not self.bibcode:
                     raise BibcodeCreationException("Failed to make a bibcode.")
+            # Use journals/issn API to get the publisher name using ISSN
+            issn_rec = []
+            issn = []
+            if self.data.get("publication").get("ISSN"):
+                issn_rec = self.data["publication"]["ISSN"]
+                for i in issn_rec:
+                    issn = i.get("issnString", None)
+                    if len(issn) == 8:
+                        issn = issn[0:4] + "-" + issn[4:]
+                    if issn:
+                        publisher = issn2info(
+                            token=self.api_token,
+                            url=self.api_url,
+                            issn=issn,
+                            return_info="publisher",
+                        )
+
+                    if publisher:
+                        publisher = str(publisher).lower()
+                        if publisher == "pnas":
+                            file_ext = publisher + ".xml"
+                        else:
+                            file_ext = publisher + "ft.xml"
+                        continue
+
+            if not issn or not publisher:
+                file_ext = self.refsource_dict.get(self.reference_source, "generic.txt")
 
-            file_ext = refsource_dict.get(self.reference_source, "generic.txt")
-            
             bibstem = self.bibcode[4:9].rstrip(".")
-            volume = self.data.get("publication", {}).get("volumeNum", "").rjust(4, '0')
-            output_dir = self.basedir + bibstem + '/' + volume
-            self.output_file = output_dir + '/' + self.bibcode + "." + file_ext
+            volume = self.data.get("publication", {}).get("volumeNum", "").rjust(4, "0")
+            output_dir = self.basedir + bibstem + "/" + volume
+            self.output_file = output_dir + "/" + self.bibcode + "." + file_ext
+            print(self.output_file)
```

Comment:
Remove print statement -- if it needs to be echoed to the user it should be a logger message or a return value.